test/fixture/setupPassport.js

```diff
@@ -14,10 +14,20 @@ passport.use(new LocalStrategy(
   }
 ));
 
-passport.serializeUser(function(user, done) {
+passport.serializeUser(function(req, user, done) {
+  if (user.name !== 'jose') {
+    return done("Invalid user", null);
+  } else if (! req) {
+    return done("Missing request", null);
+  }
   done(null, user);
 });
 
-passport.deserializeUser(function(user, done) {
+passport.deserializeUser(function(req, user, done) {
```

Comment:
Yes, this is hopelessly confusing. From what I see, this is stubbing out passport.deserializeUser, which is normally called with user object first, then request, then callback. However, a registered callback which is set by calling passport.deserializeUser with a function argument first, is pushed into a stack of deserializers, each of which is called with the request first (!). At the very least this should be documented somewhere. I'll raise a passport ticket to clarify this.